rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java

```diff
@@ -174,7 +179,11 @@ public LocalServer(Configuration conf, BookKeeperFactory bookKeeperFactory)
     @Override
     public void run()
     {
-      ExecutorService threadPool = Executors.newCachedThreadPool();
+      ExecutorService threadPool = new ThreadPoolExecutor(
+              0, getLocalTransferServerMaxThreads(conf),
+              60L, TimeUnit.SECONDS,
```

Comment:
nit: we can create config for `keepAliveTime`.

Reply:
Thread creation is fast and we're going through an idle timeout on client side anyway before the thread on server is released. So keeping this at 1m for simplicity.